internal/host/static/repository_host.go

```diff
@@ -23,44 +23,46 @@ import (
 // unique within h.CatalogId.
 func (r *Repository) CreateHost(ctx context.Context, scopeId string, h *Host, opt ...Option) (*Host, error) {
 	if h == nil {
-		return nil, fmt.Errorf("create: static host: %w", errors.ErrInvalidParameter)
+		return nil, errors.New(errors.Internal, errors.WithMsg("nil host"), errors.WithCallerOp(false))
```

Comment:
Why make this error an Internal error while the others are InvalidParameters?  

It is likely most of these should be treated as Internal errors by the API (for example: if we are not populating the catalog id for a host it should be caught prior to this point and to call the repository with a host without a catalog id defined is an error in our code).  I don't think we want to wrap all errors that will appear as internal in the API as internal for the domain api.  It feels to early to remove that information since it still needs to be returned and possibly reacted to by the caller.  Thoughts?